ui/src/app/shared/components/filter-drop-down.tsx

```diff
@@ -1,16 +1,49 @@
-import {DropDown} from 'argo-ui/src/components/dropdown/dropdown';
+import {Checkbox} from 'argo-ui';
+import * as classNames from 'classnames';
 import * as React from 'react';
-import {CheckboxList} from './checkbox-list';
+import {DropDown} from './dropdown/dropdown';
 
-export const FilterDropDown = (props: {values: {[label: string]: boolean}; onChange: (label: string, checked: boolean) => void}) => (
-    <DropDown
-        isMenu={true}
-        anchor={() => (
-            <div className={'top-bar__filter' + (props.values.size > props.values.size ? ' top-bar__filter--selected' : '')}>
-                <i className='argo-icon-filter' />
-                <i className='fa fa-angle-down' />
-            </div>
-        )}>
-        <CheckboxList values={props.values} onChange={(label, checked) => props.onChange(label, checked)} />
-    </DropDown>
-);
+interface FilterDropDownProps {
+    sections: FilterDropSection[];
+}
+
+export interface FilterDropSection {
+    title: string;
+    values: {[label: string]: boolean};
+    onChange: (label: string, checked: boolean) => void;
+}
+
+export const FilterDropDown = (props: FilterDropDownProps) => {
+    return (
+        <DropDown
+            isMenu={true}
+            anchor={() => (
+                <div className={classNames('top-bar__filter', {'top-bar__filter--selected': true})} title='Filter'>
+                    <i className='argo-icon-filter' aria-hidden='true' />
+                    <i className='fa fa-angle-down' aria-hidden='true' />
+                </div>
+            )}>
+            <ul>
+                {props.sections
+                    .filter(item => item.values)
+                    .map((item, i) => (
+                        <div key={i}>
+                            <li className={classNames('top-bar__filter-item', {title: true})}>
+                                <span>{item.title}</span>
+                            </li>
+                            {Object.entries(item.values)
+                                .sort()
+                                .map(([label, checked]) => (
+                                    <li className={classNames('top-bar__filter-item')}>
+                                        <React.Fragment>
+                                            <Checkbox id={`filter__${label}`} checked={checked} onChange={v => item.onChange(label, v)} />
+                                            <label htmlFor={`filter__${label}`}>{label}</label>
+                                        </React.Fragment>
+                                    </li>
+                                ))}
+                        </div>
+                    ))}
+            </ul>
```

Comment:
This is where we display the "filtering" aspect. It is passed as an argument (through `props.children`) to `DropDown`. None of the code here (even the code that deals with `onChange` is known/used/needed by `DropDown`, it is simply passed through and displayed.